python/GafferUI/EditScopeUI.py

```diff
@@ -195,40 +246,58 @@ def __menuDefinition( self ) :
 		if self.getPlug().getInput() is not None :
 			currentEditScope = self.getPlug().getInput().parent()
 
-		def addItem( editScope, enabled = True ) :
-
-			result.append(
-				# The underscore suffix prevents collisions with a node and
-				# it's submenu if it has nested edit scopes.
-				"/%s_" % editScope.relativeName( editScope.scriptNode() ).replace( ".", "/" ),
-				{
-					"command" : functools.partial( Gaffer.WeakMethod( self.__connectEditScope ), editScope ),
-					"active" : enabled,
-					"label" : editScope.getName(),
-					"checkBox" : editScope == currentEditScope,
-				}
-			)
-
 		if node is not None :
 			upstream = Gaffer.NodeAlgo.findAllUpstream( node, self.__editScopePredicate )
 			if self.__editScopePredicate( node ) :
 				upstream.insert( 0, node )
+
+			downstream = Gaffer.NodeAlgo.findAllDownstream( node, self.__editScopePredicate )
 		else :
 			upstream = []
+			downstream = []
+
+		# Each child of the root will get its own section in the menu
+		# if it has children. The section will be preceded by a divider
+		# with its name in the divider label.
+
+		menuHierarchy = {}
```

Comment:
I haven't got my head round all the menu-processing stuff, but let me ask a simple question : should this be `collections.OrderedDict`? I _think_ we're relying on the iteration order matching the insertion order, and that's only the case in Python 3.

Reply:
It is relying on the order being maintained - I jumped the gun on dropping Python 2 support. We're almost there! Fixed in d96244808c0f42bfecd81e9bc4610e8d1735afb9